### PR TITLE
add native parquet writer blog

### PR DESCRIPTION
**Pull request by qqibrow**
@zhenxiao @highker Please review. Thanks!

### CODE REVIEW

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+title: Native Parquet Writer for Presto
+author: Lu Niu
+authorURL: https://www.linkedin.com/in/luniu/
+---
+**Pinterest:** Lu Niu
+
+**Twitter:** Zhenxiao Luo
+
+## Overview
+With the wide deployment of Presto in more and more companies, Presto is used not only for queries, but also for data ingestion and ETL jobs. There are the needs to improve Presto’s file writer performance, especially for popular columnar file formats, e.g. Parquet, and ORC. In this article, we introduce the brand new native Parquet writer for Presto, which writes directly from Presto columnar data structure to Parquet columnar values, with a .2X to 6X throughput improvement and less cpu && memory overhead.
```
Comment (zhenxiao) [inline website/blog/2021-05-19-native-parquet-writer-for-presto.md:11]: remove . before 2X

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  
+  
+As shown in figure 2, the native Parquet writer constructs Parquet schema based on column names and types. Primitive types and complex types(Struct, Array, Map) are converted to corresponding Parquet types. Column writers are created with schema information. For each Presto page, the native writer iterates each block, transforming the Presto block into Parquet values, definition levels, and repetition levels. The corresponding column writer will write bytes streams into a Parquet file.
+  
+## Performance 
+Presto has performance testing HiveFileFormatBenchmark to test reader and writer performance. The test creates a list of pages containing millions of rows, writes them to a temp file using either native writer or the old hive record writer and then compares the performance. The following graphs show the results in three compression schemes: gzip, snappy and no compression. X-axis are various types of data; Y-axis are writing throughput. Clearly, we could see the brand new native Parquet writer outperforms the old writer. It could consistently achieve > 20% throughput improvements. The native Parquet writer performs best for BIGINT_SEQUENTIAL and BIGINT_RANDOM
```
Comment (zhenxiao) [inline website/blog/2021-05-19-native-parquet-writer-for-presto.md:36]: with GZIP compression, continue with the previous line?

**Review by zhenxiao — APPROVED** [review]
thank you, @qqibrow 
looks good to me. 2 minor issues

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+author: Lu Niu
+authorURL: https://www.linkedin.com/in/luniu/
+---
+**Pinterest:** Lu Niu
+
+**Twitter:** Zhenxiao Luo
+
+## Overview
+With the wide deployment of Presto in more and more companies, Presto is used not only for queries, but also for data ingestion and ETL jobs. There are the needs to improve Presto’s file writer performance, especially for popular columnar file formats, e.g. Parquet, and ORC. In this article, we introduce the brand new native Parquet writer for Presto, which writes directly from Presto columnar data structure to Parquet columnar values, with up to 6X throughput improvement and less cpu && memory overhead.
+
```
Comment (aweisberg) [inline website/blog/2021-05-19-native-parquet-writer-for-presto.md:12]:
```suggestion

<!-- truncate -->

```

**Review by aweisberg — CHANGES_REQUESTED** [review]
Please don't forget to squash after accepting any changes and clean up the commit message.

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+## Presto’s Old Parquet Writer
+
+As shown in Figure 1, Presto was using Hive’s old Parquet writer to write files: 
+Iterate each columnar block in a page and reconstruct every single record.
+Presto was calling the Hive record writer to consume each individual record and write value bytes to Parquet pages.
```
[the remark this answers is not in the document]
Comment (qqibrow) [reply website/blog/2021-05-19-native-parquet-writer-for-presto.md:19]: changed to `Then Presto would call`

**Comment by qqibrow** [comment]
@aweisberg Thanks for the review! fixed all comments. Please take a look. Thanks!

**Comment by qqibrow** [comment]
@aweisberg Could you take a look? thanks!

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+![Writer Throughput: GZIP](/img/blog/2021-05-19-native-parquet-writer/gzip.png)
+  
+  ## Use Native Parquet Writer in Production
+
+  The native parquet writer is ready for production deployment. To turn it on:
+```text
+# in /catalog/hive.properties
+hive.parquet.optimized-writer.enabled=true
+```
+If you have any questions, feel free to ask in the presto slack channel.
```
[the remark this answers is not in the document]
Comment (aweisberg) [reply website/blog/2021-05-19-native-parquet-writer-for-presto.md:56]: It looks like you didn't take this one? Did it not make a working link?

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+## Overview
+With the wide deployment of Presto in a growing number of companies, Presto is used not only for queries, but also for data ingestion and ETL jobs. There is a need to improve Presto’s file writer performance, especially for popular columnar file formats, e.g. Parquet, and ORC. In this article, we introduce the brand new native Parquet writer for Presto, which writes directly from Presto's columnar data structure to Parquet's columnar values, with up to 6X throughput improvement and less CPU and memory overhead.
+
+<!-- truncate -->
+
+## Presto’s Old Parquet Writer
+
+As shown in Figure 1, Presto was using Hive’s old Parquet writer to write files: 
+This old approach first iterates each columnar block in a page and reconstructing every single record.
```
Comment (aweisberg) [inline website/blog/2021-05-19-native-parquet-writer-for-presto.md:18]:
```suggestion
This old approach first iterates each columnar block in a page and reconstructs every single record.
```

File: website/blog/2021-05-19-native-parquet-writer-for-presto.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+Then Presto would call the Hive record writer to consume each individual record and write value bytes to Parquet pages.
+Whenever it exceeds the buffer size, the old writer will flush in memory data to the underlying file system. 
+
+  
+![Presto Page to Parquet RowGroups](/img/blog/2021-05-19-native-parquet-writer/page_to_parquet_rowgroups.png)
+
+  
+Presto has vectorized execution for in-memory columnar data, and Parquet is a columnar file format. The old Parquet writer was adding unnecessary overhead to convert Presto’s columnar in-memory data into row based record, and then doing one more conversion to write row based record to Parquet’s columnar on disk file format. The unnecessary data transformation adds significant performance overhead, especially when complex data types are involved such as nested structs.
```
Comment (aweisberg) [inline website/blog/2021-05-19-native-parquet-writer-for-presto.md:26]:
```suggestion
Presto has vectorized execution for in-memory columnar data, and Parquet is a columnar file format. The old Parquet writer was adding unnecessary overhead to convert Presto’s columnar in-memory data into row based records, and then doing one more conversion to write row based records to Parquet’s columnar on disk file format. The unnecessary data transformation adds significant performance overhead, especially when complex data types are involved such as nested structs.
```

**Review by aweisberg — CHANGES_REQUESTED** [review]
Two minor nits, plus the slack channel mention should link to the slack channel. Thanks!

**Review by aweisberg — APPROVED** [review]
Awesome, thank you. @zhenxiao  can you merge?

**Review by zhenxiao — APPROVED** [review]
looks good. thank you, @aweisberg

**Pull request merged**